src/core/mac/mac_filter.cpp

```diff
@@ -41,61 +41,46 @@ namespace ot {
 namespace Mac {
 
 Filter::Filter(void)
-    : mAddressMode(OT_MAC_FILTER_ADDRESS_MODE_DISABLED)
+    : mMode(kModeRssInOnly)
     , mDefaultRssIn(kFixedRssDisabled)
 {
-    for (FilterEntry *entry = &mFilterEntries[0]; entry < OT_ARRAY_END(mFilterEntries); entry++)
+    for (FilterEntry &entry : mFilterEntries)
     {
-        entry->mFiltered = false;
-        entry->mRssIn    = kFixedRssDisabled;
+        entry.mFiltered = false;
+        entry.mRssIn    = kFixedRssDisabled;
     }
 }
 
 Filter::FilterEntry *Filter::FindEntry(const ExtAddress &aExtAddress)
 {
-    FilterEntry *entry;
+    FilterEntry *rval = nullptr;
 
-    for (entry = &mFilterEntries[0]; entry < OT_ARRAY_END(mFilterEntries); entry++)
+    for (FilterEntry &entry : mFilterEntries)
     {
-        if (entry->IsInUse() && (aExtAddress == entry->mExtAddress))
+        if (entry.IsInUse() && (aExtAddress == entry.mExtAddress))
         {
-            ExitNow();
+            ExitNow(rval = &entry);
         }
     }
 
-    entry = nullptr;
-
 exit:
-    return entry;
+    return rval;
 }
 
 Filter::FilterEntry *Filter::FindAvailableEntry(void)
 {
-    FilterEntry *entry;
+    FilterEntry *rval = nullptr;
 
-    for (entry = &mFilterEntries[0]; entry < OT_ARRAY_END(mFilterEntries); entry++)
+    for (FilterEntry &entry : mFilterEntries)
     {
-        VerifyOrExit(entry->IsInUse(), OT_NOOP);
+        if (!entry.IsInUse())
+        {
+            ExitNow(rval = &entry);
+        }
```

Comment:
```suggestion
        VerifyOrExit(entry->IsInUse(), rval = &entry);
```
can we still use VerifyOrExit?

Reply:
From what I recall, we had a discussion about this (in earlier days of OT development).

For cases where we are searching within a list/array/container for an entry (while we can use `VerifyOrExit` and it would be fewer lines) the preferred approach is `"if (entry satisfies certain conditions) then exit"` which reads much more clearly (and therefore less chance of a mistake), compared to `"verify that entry does not satisfy conditions or exit"`.

`VerifyOrExit` suits well for checking and verifying that some conditions are true before continuing on with next lines of code and logic.

In general, I personally like the syntax sugars and using fewer lines wherever possible, but in this case, I also do favor readability of the style "if (check) then exit" approach.

Reply:
I agree with you that `if (condition) exit` is more readable in some loops.